lib/active_merchant/billing/gateways/ipg.rb

```diff
@@ -179,6 +181,34 @@ def add_payment(xml, payment, options = {}, credit_envelope = 'v1')
         end
       end
 
+      def add_sub_merchant(xml, submerchant)
+        xml.tag!('v1:SubMerchant') do
+          xml.tag!('v1:Mcc', submerchant[:mcc]) if submerchant[:mcc]
+          xml.tag!('v1:LegalName', submerchant[:legal_name]) if submerchant[:legal_name]
+          add_address(xml, submerchant[:address]) if submerchant[:address]
+          add_document(xml, submerchant[:document]) if submerchant[:document]
```

Comment:
Nice work getting all of these xml fields added in an easy-to-read way! 👍

Reply:
Thanks 😃